File: 824 - Goat Latin.cpp
#include <iostream>
#include <string>
#include <unordered_set>
#include <cctype>
using namespace std;

class Solution {
public:
    string toGoatLatin(string S) {
        string ap = "a";
        unordered_set<char> vowel;
        string v = "AEIOUaeiou";
        for (char c : v) {
            vowel.insert(c);
        }
        S += ' ';
        string res = "";
        string temp = "";
        char last = ' ';
        for (char c : S) {
            if (c != ' ') {
                temp += c;
            } else if (temp.size() > 0) {
                char fst = temp[0];
                if (vowel.find(fst) != vowel.end()) {
                    res += temp;
                } else {
                    res += temp.substr(1, temp.size() - 1);
                    res += fst;
                }
                res += "ma";
                res += ap;
                res += ' ';
                ap += 'a';
                temp = "";
            }
        }
        return res.substr(0, res.size() - 1);
    }
};

File: 824 - Goat Latin.py
class Solution:
    def toGoatLatin(self, S: str) -> str:
        words = S.split(' ')
        vowel = 'aeiouAEIOU'
        ap = 'a'
        for i in range(len(words)):
            if words[i][0] in vowel:
                words[i] += 'ma'
            else:
                words[i] = words[i][1:] + words[i][0] + 'ma'
            words[i] += ap
            ap += 'a'
        res = ' '.join(words)
        return res


s = Solution()
a = s.toGoatLatin("I speak Goat Latin")
print(a)
